fastapi_mcp/prompts.py

```diff
@@ -0,0 +1,277 @@
+                else:
+                    # Handle string or other simple types
+                    mcp_messages.append(
+                        types.PromptMessage(role="user", content=types.TextContent(type="text", text=str(msg)))
+                    )
+
+            return mcp_messages
+
+        except Exception as e:
+            logger.error(f"Error executing prompt '{name}': {e}")
+            raise ValueError(f"Error executing prompt '{name}': {str(e)}")
+
+    def has_prompts(self) -> bool:
+        """Check if any prompts are registered."""
+        return len(self.prompts) > 0
+
+    def auto_register_tool_prompts(self, tools: List[types.Tool], operation_map: Dict[str, Dict[str, Any]]) -> None:
+        """
+        Automatically register default prompts for each tool.
+
+        Args:
+            tools: List of MCP tools to create prompts for
+            operation_map: Mapping of operation IDs to operation details
+        """
+        for tool in tools:
+            prompt_name = f"use_{tool.name}"
+
+            # Skip if user has already registered a custom prompt with this name
+            if prompt_name in self.prompts:
+                logger.debug(f"Skipping auto-registration for {prompt_name} - custom prompt exists")
+                continue
+
+            # Generate prompt content for this tool
+            prompt_content = self._generate_tool_prompt_content(tool, operation_map.get(tool.name, {}))
+
+            # Create a simple prompt function
+            def create_tool_prompt(content: str):
+                def tool_prompt_func():
+                    return PromptMessage(role="user", content=TextContent(type="text", text=content))
+
+                return tool_prompt_func
+
+            # Register the auto-generated prompt
+            self.prompts[prompt_name] = {
+                "name": prompt_name,
+                "title": f"Usage Guide: {tool.name}",
+                "description": f"Best practices and guidance for using the {tool.name} tool effectively",
+                "arguments": [],
+                "func": create_tool_prompt(prompt_content),
+                "input_schema": {"type": "object", "properties": {}, "required": []},
+                "auto_generated": True,
+            }
+
+            logger.debug(f"Auto-registered prompt: {prompt_name}")
+
+    def _generate_tool_prompt_content(self, tool: types.Tool, operation_info: Dict[str, Any]) -> str:
+        """
+        Generate helpful prompt content for a tool.
+
+        Args:
+            tool: The MCP tool to generate content for
+            operation_info: Operation details from the operation map
+
+        Returns:
+            Generated prompt content as a string
+        """
+        # Focus on actionable guidance rather than repeating tool information
+        content_parts = [
+            f"You are about to use the **{tool.name}** tool.",
```

Comment:
can this be done once for all tools? it looks more like MCP server description, rather than tool description..